message_definitions/v1.0/development.xml

```diff
@@ -672,6 +691,24 @@
       <field type="float" name="temperature" units="K" invalid="NaN">Fuel temperature. NaN: field not provided.</field>
       <field type="uint32_t" name="fuel_type" enum="MAV_FUEL_TYPE">Fuel type. Defines units for fuel capacity and consumption fields above.</field>
     </message>
+    <message id="398" name="GENERIC_PAYLOAD_STATUS">
+      <description>Generic payload status.</description>
+      <field type="uint32_t" name="uptime_ms" units="ms">Time since the start-up of the generic payload in ms</field>
+      <field type="char[32]" name="name">Generic payload name to be used in UI. This is a NULL terminated string. If it is exactly 32 characters long, add a terminating NULL. If this string is empty, a generic name is shown to the user.</field>
+      <field type="uint16_t" name="function_count">Total number of functions on the generic payload that can be controlled.</field>
+      <field type="uint32_t" name="error_flags" enum="GENERIC_PAYLOAD_ERROR_FLAGS" display="bitmask">Errors</field>
+      <field type="uint32_t" name="custom_error_flags" display="bitmask">Bitmap used to show custom error flags.</field>
+      <field type="float" name="power_draw" units="W" invalid="NaN">The power draw of the generic payload. NaN: field not provided</field>
+      <field type="uint16_t" name="weight" units="g" invalid="0">Generic payload weight. 0: field not provided.</field>
+      <field type="float" name="temp_c" invalid="NaN">Temperature in Celsius. NaN: field not provided.</field>
+    </message>
+    <message id="399" name="GENERIC_PAYLOAD_FUNCTION_STATUS">
```

Comment:
I wonder if there's reason to expand this to include something that specifies the control scheme ie :

```
enum ControlSchemeFlags {
  LATCHING = 1,
  NON_LATCHING = 2,
  BINARY = 4, 
  ANALOG = 8, 
  PWM = 16,
};
```

The non-latching idea was motivated by the dropper needing to "reset" after actuation. However in this case the current UI only sends on USER_ACTION to deploy, not reset. (Could be updated to send RESET maybe).

But non-latching would still be interpreted by attachment, ie it could latch for a time period, then reset on its own (in the case of the dropper), or it fires and resets immediately (one shot gpio0 on/off)